### PR TITLE
Redux implementations - Cleaning code - React Loader Spinner implementations

### PR DESCRIPTION
**Pull request by lappjeff**
# Description
- Refactored forms to use Redux more efficiently 
- Removed unnecessary lines of code 
- Implemented React Loader spinners for async logic
## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

File: src/components/SubmissionForm/ProjectForm.js [hunk 1]
```diff
@@ -43,16 +42,9 @@ const ProjectForm = props => {
     e.preventDefault()
 
     let submitted_at = moment().format("MMMM Do YYYY, h:mm:ss a")
-    let display_image = state.display_image
-      ? state.display_image
-      : props.display_image
+    let display_image = props.display_image && props.display_image
```
Comment (blackmacaroon) [inline src/components/SubmissionForm/ProjectForm.js]: is this right?      props.display_image && props.display_image

Comment (lappjeff) [reply src/components/SubmissionForm/ProjectForm.js]: Yup! That's called a short circuit conditional, here's an article on it: https://codeburst.io/javascript-short-circuit-conditionals-bbc13ac3e9eb

Comment (Joshua-Disney) [reply src/components/SubmissionForm/ProjectForm.js]: I was just about to ask about this.

**Comment by lappjeff** [comment]
To ensure that a user can't submit a project missing required fields for the backend 👍

**Pull request merged**